### PR TITLE
Merging Dispatcher to ParallelDispatcher (2/2)

### PR DESCRIPTION
**Pull request by gza060625**
https://github.com/eclipse/omr/issues/4611

This is the second of two PRs in OMR.
Change all Dispatcher(D) type to ParallelDispatcher(PD)
Remove PD type cast
Change `createDispatcher()` to `createParallelDispatcher()`

Signed-off-by: Enson Guo <enson.guo@ibm.com>

### CODE REVIEW

**Comment by gza060625** [comment]
@amicic Please review.

**Comment by amicic** [comment]
I believe there are more explicit (MM_ParallelDispatcher *) casts that now can be removed. I counted 3 being removed, but 8 being introduced by https://github.com/eclipse/omr/pull/5407.

Bottom line, there should be no more of those casts whether recently introduced or potentially existing/old ones.

**Comment by gza060625** [comment]
@amicic Yes, you are right. I have deleted 6 more casts. Right now, only one cast left (not introduced by me and is necessary).

File: gc/base/ParallelDispatcher.hpp [hunk 1]
```diff
@@ -35,13 +35,18 @@
...
 #include "GCExtensionsBase.hpp"
 
 class MM_EnvironmentBase;
 
-class MM_ParallelDispatcher : public MM_Dispatcher
+/**
+ * @todo Provide define documentation
+ * @ingroup GC_Base_Core
+ */
+#define J9MODRON_HANDLE_NEXT_WORK_UNIT(envPtr) envPtr->_currentTask->handleNextWorkUnit(envPtr)
```
Comment (amicic) [inline gc/base/ParallelDispatcher.hpp:47]: I know this was in Dispatcher, but semantically it belongs more to (Parallel)Task - it's just a macro-wrapper of Task's method handleNextWorkUnit. Try to move it there.

**Comment by gza060625** [comment]
@youngar This PR is ready for further review. Thanks.

**Comment by amicic** [comment]
@fjeremic if you have time please try to review/merge. While touches a lot of files, it's a low risk, a non-functional change.

**Comment by fjeremic** [comment]
@genie-omr build all

**Pull request merged**